### ISSUE TITLE
Generalize OptionsPattern inspection

### ISSUE THREAD

**Issue by szhorvat** [WolframResearch/codeinspector#2]
The following reports a potential problem as expected:

```
CodeInspectSummarize[
 "f[x_ : {}, OptionsPattern[]] := x"
 ]
```

![image](https://user-images.githubusercontent.com/1212871/80234702-e37bb200-8658-11ea-9808-77356e9ff33b.png)

This one should too:

```
CodeInspectSummarize[
 "f[x_List : {}, OptionsPattern[]] := x"
 ]
```

Currently, it does not report anything.

`List` is not sufficiently specific and the same issues apply as with a naked `_`. `OptionsPattern[]` will eat `List`s.

To be clear, I am not asking for handling all cases that are not sufficiently specific. That is both unreasonable and impossible. I am asking just for `_List`, which should cover a significant fraction of cases that remain after having dealt with a naked `_`.

**Comment by bostick** [comment]
Good catch!

**Comment by bostick** [comment]
This is fixed in the latest sources
